### DISCUSSION TITLE
Website with auto generated index files?

### DISCUSSION THREAD

**Discussion by Wysardry** [getnikola/nikola#3681] [Q&A]
Is it possible to create a website (not blog) with Nikola, without having to create and modify the home or category index pages manually?

I would like to just add code for articles to the correct categories and have Nikola create the index pages (complete with links and descriptions) for me - as would happen with a blog - but without any tag or archive pages being produced.

The categories would be in directories off the root directory, not within a categories directory. For example, domain/cars/ not domain/categories/cars/. Date info would not be included in URLs.

I'm completely new to Python and Nikola, but do know some HTML, CSS and Perl, so please bear this in mind when answering.

I have read "Creating a Site (Not a Blog) with Nikola," but that seems to expect the pages to be unrelated to each other and for you to create index pages manually.

**Comment by Kwpolska** [comment]
Thi kinda sounds like you want a blog, but with some features disabled. Those settings might produce what you want:

```py
DISABLED_PLUGINS = ["classify_tags", "classify_archive", "classify_authors"]
CATEGORY_PATH = ""
CATEGORY_PREFIX = ""
CATEGORIES_INDEX_PATH = "categories.html"
```

**Comment by Wysardry** [comment]
I'm not really sure if removing fatures from posts or adding features to pages would be the right approach to take.

It would make the most sense to me if I could add a third customised document type called "articles" (or whatever), but I have no idea whether that is possible.

You suggestion looks like a simple change, but I would want the categories listed on the home page. Would "index.html" work instead of "categories.html" or would that cause conflicts with other features that try to write to the home page?

**Reply by Kwpolska** [reply]
You can't add a new type besides posts or pages to Nikola. Removing features from posts is easier than adding features to pages.

If you want to set `CATEGORIES_INDEX_PATH = "index.html"`, you would need to move (set `INDEX_PATH`) or disable (`DISABLED_PLUGINS = […, "classify_indexes"]`) the chronological post index.

**Comment by Wysardry** [comment]
I was hoping to reserve the option to add a blog section later (so three page types would make more sense), but I could probably use a separate site project for that if needed.

What I am hoping to do is set up a site that has categories and their descriptions listed on the home page, possibly with a few links and descriptions for featured or most recently updated articles.

Categories would be directories off the site root, each with an index page containing links and descriptions for the articles in that category, preferably sorted by when they were most recently updated (rather than the more usual publishing date) or by weight. I would prefer that these were automatically generated or it wouldn't be much quicker than using an HTML editor with a templating system.

There would not be any archives, as the articles on a particular topic would be updated or expanded rather than new posts being made on a regular basis. There would only be one URL per article in the form of site-name/category-name/article-name.html

At a later time I would also like to create another site on a more complex topic where articles would be be stored in site-name/category-name/subcategory-name/article-name.html if possible.

Is Nikola suitable for this, or should I be looking more closely at Pelican or MkDocs instead?

**Reply by Kwpolska** [reply]
Nikola could probably do this, although it may take some work to get this done. Your index page could be built using a bunch of [post lists](https://getnikola.com/handbook.html#post-list), one per category. Post lists can display pages if needed. It should also be possible to sort by the `updated` field (but in your case, it might make sense to just use the `date` field and put the last update date there). Pages can be placed anywhere (Nikola will reproduce the directory layout from input); categories can have hierarchies.

The features you don’t need (archives and blog-style indexes) can be disabled (as they are just plugins).

**Comment by Wysardry** [comment]
Thanks. I'm not too concerned about it requiring a lot of effort if it will save me having to update index pages manually every time I update an article.

Would I be able to change the format of the produced post lists? I'm guessing they are HTML LI lists by default? I would like them displayed in a DL, with the article title (linked) as the DT and the article description as a DD.

I would much prefer to use both the date and the updated fields, as I would like returning visitors to know whether a particular article is completely new or just updated.

**Reply by Kwpolska** [reply]
> Would I be able to change the format of the produced post lists? I'm guessing they are HTML LI lists by default? I would like them displayed in a DL, with the article title (linked) as the DT and the article description as a DD.

Sure, you can provide a custom template.
